lib/DelegatingSerializedCache.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Amp\Cache;
+
+use Amp\Failure;
+use Amp\Promise;
+use Amp\Sync\SerializationException;
+use Amp\Sync\Serializer;
+use function Amp\call;
+
+final class DelegatingSerializedCache implements SerializedCache
```

Comment:
I don't think we need an interface here. A single class `SerializedCache` should be fine.

Reply:
If `AtomicCache` is to use a `SerializedCache`, then we need an interface.

Reply:
I don't see how `AtomicCache` is related here, could you clarify?

Reply:
I originally planned to rebase #8 on this, so `AtomicCache` would take an instance of `SerializedCache` and implement `SerializedCache`.

I'm thinking though that since we can't extend `Cache`, I'd rather keep `AtomicCache` as-is.